Source/Reporters/JunitXMLReporter.jsl

```diff
@@ -0,0 +1,247 @@
+
+				info_types = test_case_info[k] << get keys;
+				
+				test_case_content = "";
+				For(l=1, l<=NItems(test_case_info[k]), l++,
+					info_type = info_types[l];
+					props = Associative Array();
+					props["type"] = info_type;
+					failure_content = this:generate_xml_tag("failure", test_case_info[k][info_type], props);
+
+					test_case_content ||= failure_content;
+
+				);
+				props = Associative Array();
+				props["name"] = test_case_names[k];
+				test_case_xml ||= this:generate_xml_tag("testcase", trim(test_case_content), props);
+				
+			);
+			num_tests = NItems(test_case_names);
+			props = Associative Array();
+			props["name"] = test_suites[i];
+			props["tests"] = Char(num_tests);
+
+			// find test results that are non empty
+			num_successes = Nitems(loc(test_case_info, [=>]));
+			props["failures"] = Char(num_tests-num_successes);
+			test_suite_tag = this:generate_xml_tag("testsuite", trim(test_case_xml), props);
+			content ||= test_suite_tag;
+		);
+		
+		props = Associative Array();
+		date = today();
+		props["id"] = format(date, "yyyymmdd" ) || "_" || Char(Hour(date)) || Char(Minute(date)) || Char(Second(date));
+		props["tests"] = Char(Nitems(this:successes) + NItems(this:failures) + NItems(this:unexpected throws));
+		props["failures"] = Char(NItems(this:failures) + NItems(this:unexpected throws));
+
+		xml_result = start_line || this:generate_xml_tag("testsuites", trim(content), props);;
+		Return(xml_result);
+	);
+	
+	// Function: show report
+	//
+	// Dump a report to the log detailing all the successes and
+	// failures for each test in JunitXML format shown in the comments at the top.
+	show report = Method( {},
+		num successes = Length( this:successes );
+		num failures = Length( this:failures );
+		num unexpected throws = Length( this:unexpected throws );
+
+		hash_results = Associative Array();
+
+		For( i = 1, i <= num successes, i++,
+			char test expr = If( Is Expr( Name Expr( this:successes[i][2] ) ),
+				Char( Name Expr( this:successes[i][2] ) ),
+				Char( this:successes[i][2] )
+			) || " ";
+			tmp_label = this:successes[i][1];
+			list_tmp_label = words(tmp_label, "⮚");
+			curr_suite = Trim(list_tmp_label[1]);
+			curr_test = Trim(list_tmp_label[2]) || "_" || Trim(list_tmp_label[3]);
+			// empty for a success
+			success_hash = Associative Array();
+			If(Contains(hash_results << get keys, curr_suite),
+				hash_results[curr_suite][curr_test] = Associative Array();
+				,
+				hash_results[curr_suite] = Associative Array(Eval List({curr_test}), Eval List({success_hash}));
+			);
+		);
+
+		// associative array of the test results
+		// test suite => test case => successes, failures, throws
+		
+		For( i = 1, i <= num failures, i++,
+			char test expr = If( Is Expr( Name Expr( this:failures[i][2] ) ),
+				Char( Name Expr( this:failures[i][2] ) ),
+				Char( this:failures[i][2] )
+			) || " ";
+			tmp_label = this:failures[i][1];
+			list_tmp_label = words(tmp_label, "⮚");
```

Comment:
Unfortunately, if the `ut assert that` is run outside of a `ut test`, there may not be three elements in this `list_tmp_label`. See [UtDetailsTreeReporter](https://github.com/sassoftware/jsl-hamcrest/blob/289124e570371580fe2ad95f879284e7d0740139/Source/Addin/Reporters.jsl#L280-L289) and #59.

Reply:
Should be handled in commit b0d89cefc383a60df3a7732de648366d8bd621cd. I still need to handle labels that could be 4 elements, but now 1 and 3 should work fine. The main idea of the solution is to put assertions outside of test cases in an anonymous test suite and numbered anonymous test cases if no label is supplied at all.

Reply:
4 part labels are also handled now